src/components/links/SearchBar.tsx

```diff
@@ -2,20 +2,45 @@ import Image from 'next/image';
 import searchIcon from '@/assets/icons/ic_search.svg';
 import closeIcon from '@/assets/images/close.png';
 
-export const SearchBar = () => {
+type SearchSubmit = (e: React.FormEvent<HTMLFormElement>) => void;
+
+interface SearchBarProps {
+  searchText: string;
+  setSearchText: React.Dispatch<React.SetStateAction<string>>;
+  searchSubmit: SearchSubmit;
+}
+
+export const SearchBar = ({
+  searchText,
+  setSearchText,
+  // setSearchValue,
+  searchSubmit,
+}: SearchBarProps) => {
+  const searchDelete = () => {
+    setSearchText('');
+  };
+
   return (
     <div className="m-auto max-w-[1060px] h-auto px-4 py-[15px] flex justify-between items-center rounded-[10px] bg-[#f5f5f5]">
-      <div className="w-full flex justify-center items-center gap-3">
-        <Image src={searchIcon} alt="검색 아이콘" width={16} height={16} />
-        <input
-          className="w-full focus:outline-none bg-[#f5f5f5] placeholder-[#666666] font-[Pretendard] not-italic leading-[24px]"
-          type="text"
-          placeholder="링크를 검색해 보세요"
-        />
-      </div>
-      <button type="button">
-        <Image src={closeIcon} alt="삭제 아이콘" width={24} height={24} />
-      </button>
+      <form className="w-full" onSubmit={searchSubmit}>
+        <div className="w-full flex justify-center items-center gap-3">
+          <Image src={searchIcon} alt="검색 아이콘" width={16} height={16} />
+          <input
+            className="w-full focus:outline-none bg-[#f5f5f5] placeholder-[#666666] font-[Pretendard] not-italic leading-[24px]"
+            type="text"
+            value={searchText}
+            placeholder="링크를 검색해 보세요"
+            onChange={(e) => {
+              setSearchText(e.target.value);
+            }}
```

Comment:
핸들러를 따로 선언해서 타입 지정을 해주는건 어떠신가요??
타입스크립트를 쓰면 타입 지정을 통해 가독성과 디버깅에 도움을 주는 장점이 있는데 이 장점이 사라지는 것 같아보여요!

Reply:
상위 컴포넌트에서 핸들러를 만들어서 전달해주는 것도 좋은 것 같습니다!
```suggestion
// 상위 컴포넌트
const handleSearchText = (e:React.ChangeEvent<HTMLInputElement>) => {
  const target = e.target as HTMLInputElement;
  setSearchText(target.value);
}
// 하위 컴포넌트: SearchBar.tsx
interface SearchBarProps {
  // ...
  handleSearchText: (e:React.ChangeEvent<HTMLInputElement>) => void;
}
// ...
            onChange={handleSearchText}
```